azalea/src/nearest_entity.rs

```diff
@@ -0,0 +1,181 @@
+use azalea_entity::Position;
+use azalea_world::{InstanceName, MinecraftEntityId};
+use bevy_ecs::{
+    prelude::Entity,
+    query::{ReadOnlyWorldQuery, With},
+    system::{Query, SystemParam},
+};
+
+/// This system parameter can be used as a shorthand for quickly finding an
+/// entity, (or several) close to a given position.
+///
+/// This system parameter allows for additional filtering of entities based off
+/// of ECS marker components, such as `With<>`, `Without<>`, or `Added<>`, etc.
+/// All functions used by this system parameter instance will respect the
+/// applied filter.
+///
+/// ```no_run
+/// /// All bots near aggressive mobs will scream in chat.
+/// pub fn bots_near_aggressive_mobs(
+///     bots: Query<Entity, With<Bot>>,
+///     entity_finder: EntityFinder<With<Aggressive>>,
```

Comment:
`Aggressive` is a metadata thing that's only used for like aggressive wolves and stuff iirc, this should probably just use `AbstractMonster` instead.